### PR TITLE
Adds the concept of "padding" to BlockSvg

### PR DESCRIPTION
**Pull request by Hamms**
BlockSvgFramed adds a frame around the block that extends into the
negative space, as well as below the block itself. This causes some
problems, most notably for the automatic positioning of start blocks, as
the visual space of the block is both taller than and "starts" "above"
the top of the block itself.

This PR adds a getPadding method to BlockSvg (and BlockSvgFramed) and
teachs the xml block layout logic how to use it to appropriately
position padded blocks.

| Before | After |
| --- | --- |
| image | image |

fixes https://www.pivotaltracker.com/n/projects/1256652/stories/106762696

### CODE REVIEW

**Comment by Hamms** [comment]
@joshlory could you take a look at this?

File: blockly-core/core/ui/block.js [hunk 1]
```diff
@@ -558,6 +558,14 @@ Blockly.Block.prototype.getBox = function() {
 };
 
 /**
+ * Returns the padding of the SVG or null if none exists
+ * @return {Object} object with padding values for top, bottom, left, and right
+ */
+Blockly.Block.prototype.getSvgPadding = function() {
+  return this.svg_ && this.svg_.getPadding();
```
Comment (joshlory) [inline blockly-core/core/ui/block.js:565]: Better to return `{top: 0, right: 0, bottom: 0, left: 0}` vs. `false` when `this.svg_` is null?  `padding.*` is referenced below without checking if `padding` is null.

Comment (Hamms) [inline blockly-core/core/ui/block.js:565]: I went back and forth on this for a while; my ultimate thinking was that SVG padding doesn't make sense at all if SVG is null, and that I wanted the structure of the `padding` object to be defined by BlockSvg.

You're right, though; I should totally have a null check below

**Comment by joshlory** [comment]
LGTM after adding null check.

**Pull request merged**